packages/apps/kadena-docs/src/components/Layout/components/SideMenu/SideMenu.tsx

```diff
@@ -11,23 +11,26 @@ import {
 } from './styles';
 import { useSideMenu } from './useSideMenu';
 
+import { IMenuItem } from '@/types/Layout';
 import React, { FC } from 'react';
 
 interface IProps {
   closeMenu: () => void;
+  menuItems: IMenuItem[];
 }
 
-export const SideMenu: FC<IProps> = ({ closeMenu }) => {
+export const SideMenu: FC<IProps> = ({ closeMenu, menuItems }) => {
   const {
     active,
     menuRef,
     subMenuRef,
     clickMenu,
     clickSubMenu,
-    hasSubmenu,
     setActive,
-    activeTitle,
-  } = useSideMenu(closeMenu);
+    activeItem,
+  } = useSideMenu(closeMenu, menuItems);
+
+  const hasSubmenu = activeItem?.children.length ?? 0;
```

Comment:
Can we make this a boolean and ditch the `> 0` below? Or do we need the length for something else? Then it might be better to rename it to `submenuLength` or something similar?